### DISCUSSION TITLE
Add support for solidjs

### DISCUSSION THREAD

**Discussion by gabrieldemian** [wevm/wagmi#1486] [Ideas]
Hello!
My company wants to use Solidjs for new projects, but most web3 libraries only support React.
That being said, can I start to work on a PR to add Solidjs support?

**Comment by tmm** [comment]
Definitely interested in a SolidJS adapter at some point. Right now, the core team is focused on tightening up a few things on the roadmap. I would recommend playing around with it as a community project and I can follow up on this thread when we are ready for a PR in wagmi.

**Reply by gabrieldemian** [reply]
Amazing, I cloned the project and I made an adaptor for Solid, I only have the accounts hooks so far, but it is working with all the reactivity of Solid. Please do let me know when it is a good time for me to share my code. Happy new year!

**Reply by NeOMakinG** [reply]
@gabrieldemian I would be interested in looking at your code because I started too in the past and accounts hooks require a lot of work, tanstack `solid-query` is a bit different and `useQueryErrorResetBoundary` react's hook hasn't been implemented, I migrated 4-5 hooks on that part with utils and dependencies

**Reply by gabrieldemian** [reply]
@NeOMakinG I didn't had time to integrate solid-query yet. You can use signals for the hooks args, and they will react normally. [here is the repo](https://github.com/gabrieldemian/wagmi-solid). I changed a lot of things, because I was studying some things while I was hacking with Solid. Run `yarn dev` and you will have a solid-start running on localhost.

**Reply by NeOMakinG** [reply]
Thanks for your response, in case you want to get a look at what I started (It's not really usable right now, it would require some tweaks): https://github.com/NeOMakinG/wagmi/tree/solid/packages/solid

Not sure that I'll continue but maybe we could collab, my vision on the thing was to stick as much as possible to the React codebase so it would be easier to maintain in the future

**Reply by gabrieldemian** [reply]
thank you! I'm available to work on this on my free time. I will clone the repo again and only make the necessary changes, and also try to add solid-query, let me know if you are available to colab.

**Reply by NeOMakinG** [reply]
> thank you! I'm available to work on this on my free time. I will clone the repo again and only make the necessary changes, and also try to add solid-query, let me know if you are available to colab.

Followed you on twitter so we can talk :)

**Reply by glitch-txs** [reply]
any progress on this? @NeOMakinG @gabrieldemian

**Reply by gabrieldemian** [reply]
unfortunately not, we did open this PR a long time ago https://github.com/wevm/wagmi/pull/1947 but it got closed, if you have enough free time I would recommend you to build your own hooks for your own library @glitch-txs

**Reply by glitch-txs** [reply]
Seems like with Wagmi v2 there's need of some re-writing. If we get a repo mantained an running I can help on getting it into Web3Modal documentation. We'll probably be adding the Vue one from @unicape as well

**Reply by gabrieldemian** [reply]
@glitch-txs sounds good, I can contribute to the new repo. Do you mean a repo that is independent of wagmi, from scratch?

**Reply by glitch-txs** [reply]
nop, a fork like you have, but up to date, Web3Modal works with `@wagmi/core` so anything on top of that will work as well :)

**Reply by glitch-txs** [reply]
also it would be nice to potentially revisit it merging it into main here I guess

**Reply by gabrieldemian** [reply]
if you plan to merge it here I guess it's best to start really small, with just 1 hook to make it easy to review and merge, after that we can be more ambitious, feel free to send me a msg on discord or linkedin for easy communication: @einezwei

**Reply by glitch-txs** [reply]
Do you have twitter? Seems like we can build something check https://twitter.com/awkweb/status/1753483396825976949?t=ChYmzIb2t7-5tFA9LnXnbQ&s=19

**Comment by nienow** [comment]
Just wondering if you have made any progress on this. I was looking for a solidjs web3 library, and thinking of writing my own.

**Reply by gabrieldemian** [reply]
yes. we (me and @NeOMakinG) had created the integration with Solid in Wagmi, two times. Everything was working fine, but the devs did not merge our PRs. I recommend you to build your own library from scratch

**Reply by NeOMakinG** [reply]
@nienow https://github.com/wagmi-dev/wagmi/pull/1947

This PR is waiting for the core team to get some attention after the viem integration

**Reply by llllvvuu** [reply]
We're on `wagmi@1.3.3` now, so `viem` integration is stable. The PR will have to be migrated through quite a few breaking changes https://wagmi.sh/react/migration-guide but I had managed to migrate a pretty large project within a day so it should be doable for this.

IDK if you guys are still maintaining that branch @gabrieldemian @NeOMakinG . I should also have some time to help out later in the year if this is still WIP then. FWIW each `packages/` should be publishable to npm directly so we could always publish it independently from `wagmi` team if we were so inclined.

**Reply by NeOMakinG** [reply]
We didn't touch anything on the branch since @tmm told us that we had to wait for viem to be implemented 

The branch shouldn't be that hard to rebase honestly, I'm just waiting for a go from the team to rebase it.

If you want to rebase it and publish it on npm, feel free I pretty sure we can give you the access to the fork so we don't duplicate the work ✌️

**Comment by unicape** [comment]
@NeOMakinG It is recommended to maintain it independently of wagmi first and wait for the wagmi team to have a plan to provide review before merging. The `Vue` version of [wagmi](https://github.com/unicape/use-wagmi) has been used for quite some time

**Comment by joaofranciscoguarda** [comment]
Any update on this? Wagmi 2.0 came out and I was thinking about making my version of wagmi-solid, but then I thought if someone already got that ideia, and here I'm, didnt find any package, tried my self but I'm struggling to see how to test the package and maintain

**Comment by SamuelLHuber** [comment]
Would love to know if there's an easy solution here. I'd love to not write a custom wagmi port to solid
